src/Routes/DetailPage/DetailPage.tsx

```diff
@@ -82,26 +84,38 @@ const DetailPage = () => {
     setIsKebabOpen(!isKebabOpen);
   };
 
+  const OnShowConfirmDelete = () => {
+    setIsOpenConfirmDelete(true);
+  };
+
+  const onDismissConfirmDelete = () => {
+    setIsOpenConfirmDelete(false);
+  };
+
+  const onDelete = (domain?: Domain) => {
+    if (domain?.domain_id !== undefined) {
+      const domainId = domain.domain_id;
+      resources_api
+        .deleteDomain(domainId)
+        .then((response) => {
+          if (response.status == 204) {
+            navigate('/domains', { replace: true });
+          } else {
+            // TODO show-up notification with error message
+          }
+        })
+        .catch((error) => {
+          // TODO show-up notification with error message
+          console.log('error onClose: ' + error);
+        });
+    }
+  };
+
   const dropdownItems: JSX.Element[] = [
     <DropdownItem
       key="delete"
-      onClick={(value) => {
-        console.log('Deleting domain: ' + value);
-        if (domain_id !== undefined) {
-          resources_api
-            .deleteDomain(domain_id)
-            .then((res) => {
-              if (res.status === 204) {
-                console.info('Domain ' + value + ' was deleted');
-                appContext?.deleteDomain(domain_id);
```

Comment:
Will rather ask. `appContext?.deleteDomain(domain_id);` is not part of the new `onDelete.` Is it intentional?

Missing it is not a big deal. There is a period of time after redirection when the deleted domain is still visible, but it gets corrected automatically when the domains list page refreshes its content. The refresh took some time in my env when I had there 12 domains.

Reply:
thanks! good catch! I didn't realize! :+1: 

I am updating the code.

Reply:
Thanks - LGTM again.